mitosheet/mitosheet/public/v3/sheet_functions/bool_functions.py

```diff
@@ -123,6 +124,76 @@ def IF(condition: pd.Series, true_series: AnyPrimitiveOrSeriesInputType, false_s
+                "element": "boolean_condition",
+                "description": "An expression or series that returns True or False values. Valid conditions for comparison include ==, !=, >, <, >=, <=."
+            },
+            {
+                "element": "value_if_true",
+                "description": "The value the function returns if condition is True."
+            }, ...
+        ]
+    }
+    """
+    # Go through all of the conditions. The first condition that is true for
+    # *each cell* should be selected. 
+    results = None
+    for index, condition in enumerate(argv):
+        # The args are alternating between the conditions and the values corresponding to them.
+        # If this is a series of booleans in the even index, we assume it is the condition.
+        # We skip the values in this loop because they're only used as a function of the conditions. 
+        if index % 2 == 0:
+            if isinstance(condition, pd.Series) and condition.dtype == bool:
+                # For each cell, check if the value is true.
+                # If it is, use the "true_series" to fill the value in the result series.
+                true_series = get_series_from_primitive_or_series(argv[index+1], condition.index)
+                # Add the new data into the series
+                new_series = true_series[condition]
+                if results is None:
+                    results = new_series
+                else:
+                    # Combine the series to fill in the None values in the final series
+                    results = results.combine_first(new_series)
+            # This is the "else" case for the IFS formula.
+            elif condition == True:
+                base_index = next(iter(s.index for s in argv if isinstance(s, pd.Series)), None)
+                if base_index is None:
+                    error = MitoError(
+                        'invalid_args_error',
+                        'IFS',
+                        f"IFS requires at least one boolean series condition.",
+                        error_modal=False
+                    )
+                    raise error
+                # If the condition is just True, we can just use the "true_series" to fill the value in the result series.
+                true_series = get_series_from_primitive_or_series(argv[index+1], base_index)
+                if results is None:
+                    results = true_series
+                else:
+                    # Combine the series to fill in the None values in the final series
+                    results = results.combine_first(true_series)
+            # If this is an even indexed argument, it should be a boolean series or True. If it isn't, we throw an error.
+            else:
+                error = MitoError(
+                    'invalid_args_error',
+                    'IFS',
+                    f"IFS requires a boolean series condition as the first argument, but got {condition} instead.",
```

Comment:
Nice -- user-facing error messages! Since our users are new to Python, they don't know what series are -- also the {condition} might be a huge series, and so this error message would be insanely long. 

Can you try and think of another error message that would be more helpful to our users? Hint -- talking about the "5th" argument to the function is something users get. They also probably get the term "boolean".

Reply:
Updated! How does that look?